mypy/test/testconfigparser.py

```diff
@@ -1,168 +1,50 @@
-                    [mypy]
-                    files =  file1.py ,   file2.py  ,   file3.py
-                    """
-                )
-
-            options = Options()
-
-            parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
-
-            self.assertEqual(options.files, ["file1.py", "file2.py", "file3.py"])
-
-    def test_parse_config_file_with_empty_files_key(self) -> None:
-        """An empty files key should result in an empty list."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
-
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files =
-                    """
-                )
-
-            options = Options()
-
-            parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
-
-            self.assertEqual(options.files, [])
-
-    def test_parse_config_file_with_only_comma(self) -> None:
-        """A files key with only a comma should raise an error."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
-
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files = ,
-                    """
-                )
-
-            options = Options()
-
-            with self.assertRaises(ValueError) as cm:
-                parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
-
-            self.assertIn("Invalid config", str(cm.exception))
-
-    def test_parse_config_file_with_only_whitespace(self) -> None:
-        """A files key with only whitespace should result in an empty list."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
-
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files =
-                    """
-                )
-
-            options = Options()
-
-            parse_config_file(options, lambda: None, config_path, stdout=None, stderr=None)
-
-            self.assertEqual(options.files, [])
-
-    def test_parse_config_file_with_mixed_valid_and_invalid_entries(self) -> None:
-        """Mix of valid and invalid filenames should raise an error."""
-        with tempfile.TemporaryDirectory() as tmpdirname:
-            config_path = os.path.join(tmpdirname, "test_config.ini")
-
-            with open(config_path, "w") as f:
-                f.write(
-                    """
-                    [mypy]
-                    files = file1.py, , , file2.py
-                    """
-                )
+@pytest.mark.parametrize(
+    "config_content, expected_files, expected_exception",
```

Comment:
It's usually nicer to have an iterable of strings for the params list. Sparse structures read better.
```suggestion
    ("config_content", "expected_files", "expected_exception"),
```